src/user/library/index.ts

```diff
@@ -0,0 +1,209 @@
+import {AdSetForm, CampaignForm, Creative, OS, Segment} from "../views/adsManager/types";
+import {
+  CreateAdInput,
+  CreateAdSetInput,
+  CreateCampaignInput,
+  CreativeInput, GeocodeInput, UpdateAdSetInput, UpdateCampaignInput
+} from "../../graphql/types";
+import axios from "axios";
+import {print} from "graphql";
+import {CreateCreativeDocument} from "../../graphql/creative.generated";
+import {CampaignFragment} from "../../graphql/campaign.generated";
+import {CreateAdDocument} from "../../graphql/ad-set.generated";
+
+export async function transformNewForm(
+  form: CampaignForm,
+  auth: any,
+  advertiserId: string
+): Promise<CreateCampaignInput> {
+  const adSets = form.adSets;
+  const transformedAdSet: CreateAdSetInput[] = [];
+
+  for (let adSet of adSets) {
+    const ads: CreateAdInput[] = [];
+
+    for (let ad of adSet.creatives) {
+      const creative = await transformCreative(
+        ad,
+        adSet,
+        auth.accessToken,
+        advertiserId
+      );
+      ads.push(creative);
+    }
+
+    const base: CreateAdSetInput = {
+      billingType: adSet.billingType,
+      execution: "per_click",
+      perDay: 1,
+      segments: adSet.segments.map((s) => ({ code: s.code, name: s.name })),
+      oses: adSet.oses,
+      totalMax: 10,
+      conversions: adSet.conversions,
+      ads: ads,
+    }
+
+    transformedAdSet.push(base);
+  }
+
+  return {
+    currency: form.currency,
+    dailyCap: form.dailyCap,
+    dailyBudget: form.dailyBudget,
+    endAt: form.endAt,
+    geoTargets: form.geoTargets.map((g) => ({ code: g.code, name: g.name })),
+    name: form.name,
+    advertiserId: advertiserId,
+    externalId: "",
+    format: form.format,
+    source: "direct",
+    startAt: form.startAt,
+    state: form.state,
+    type: form.type,
+    budget: form.budget,
+    adSets: transformedAdSet,
+  }
+}
+
+async function transformCreative(
+  creative: Creative,
+  adSet: AdSetForm,
+  accessToken: string,
+  advertiserId: string,
+): Promise<CreateAdInput> {
+  const notification: CreativeInput = {
+    advertiserId: advertiserId,
+    name: creative.name,
+    payloadNotification: { title: creative.title, body: creative.body, targetUrl: creative.targetUrl },
+    state: "under_review",
+    type: {
+      code: "notification_all_v1",
+    }
+  }
+  const withId = await createNotification(accessToken, notification);
+  return {
+    state: "under_review",
+    webhooks: [],
+    creativeId: withId,
+    prices: [{
+      amount: adSet.price,
+      type: adSet.billingType === "cpc" ? "click" : "view"
+    }]
+  };
+}
+
+// TODO: Get rid of this ASAP. Currently necessary because when creating a campaign, you need existing creativeId.
+async function createNotification(accessToken: string, createInput: CreativeInput) {
```

Comment:
creates notification creative, currently only type supported by ads-admin. Had to do it this way because backend does not support passing in new creatives